pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -263,32 +264,36 @@ func (c *Reconciler) removeFinalizer(s *v1alpha1.CloudAuditLogsSource) {
 	s.Finalizers = finalizers.List()
 }
 
-func (c *Reconciler) updateStatus(ctx context.Context, desired *v1alpha1.CloudAuditLogsSource) (*v1alpha1.CloudAuditLogsSource, error) {
-	source, err := c.auditLogsSourceLister.CloudAuditLogsSources(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, err
-	}
-	// Check if there is anything to update.
-	if equality.Semantic.DeepEqual(source.Status, desired.Status) {
-		return source, nil
-	}
-	becomesReady := desired.Status.IsReady() && !source.Status.IsReady()
-
-	// Don't modify the informers copy.
-	existing := source.DeepCopy()
-	existing.Status = desired.Status
-	src, err := c.RunClientSet.EventsV1alpha1().CloudAuditLogsSources(desired.Namespace).UpdateStatus(existing)
-
-	if err == nil && becomesReady {
-		duration := time.Since(src.ObjectMeta.CreationTimestamp.Time)
-		logging.FromContext(ctx).Desugar().Info("CloudAuditLogsSource became ready", zap.Any("after", duration))
-		c.Recorder.Event(source, corev1.EventTypeNormal, "ReadinessChanged", fmt.Sprintf("CloudAuditLogsSource %q became ready", source.Name))
-		if err := c.StatsReporter.ReportReady("CloudAuditLogsSource", source.Namespace, source.Name, duration); err != nil {
-			logging.FromContext(ctx).Infof("failed to record ready for CloudAuditLogsSource, %v", err)
+func (c *Reconciler) updateStatus(ctx context.Context, existing *v1alpha1.CloudAuditLogsSource, desired *v1alpha1.CloudAuditLogsSource) error {
+	existing = existing.DeepCopy()
```

Comment:
this sounds confusing. Can you just pass `original`, and then do existing := original.DeepCopy() ?
here and in other places as well

Reply:
done